samza-test/src/main/java/org/apache/samza/test/framework/system/CollectionStreamSystemSpec.java

```diff
@@ -32,11 +33,17 @@
  * Following system level configs are set by default
  * <ol>
  *   <li>"systems.%s.default.stream.samza.offset.default" = "oldest"</li>
+ *   <li>"jobs.job-name.systems.%s.default.stream.samza.offset.default" = "oldest"</li>
  *   <li>"systems.%s.samza.factory" = {@link InMemorySystemFactory}</li>
+ *   <li>"jobs.job-name.systems.%s.samza.factory" = {@link InMemorySystemFactory}</li>
  * </ol>
- *
+ * The "systems.*" configs are required since the planner uses the system to get metadata about streams during
+ * planning. The "jobs.job-name.systems.*" configs are required since configs generated from user provided
+ * system/stream descriptors override configs originally supplied to the planner. Configs in the "jobs.job-name.*"
+ * scope have the highest precedence.
  */
 public class CollectionStreamSystemSpec {
```

Comment:
Why do we still need this SystemSpec for CollectionStreamSystem? Can we replace it with CollectionStreamSystemDescriptor?

Reply:
Yup, @Sanil15 is going to do that in a follow up PR.

Reply:
We will call it InMemorySystemDescriptor to be consistent with naming, corresponding streams will be called InMemoryInputDescriptor and InMemoryOutputDescriptor